lib/queue.js

```diff
@@ -423,6 +430,7 @@ Queue.prototype.run = function(concurrency){
   var promises = [];
   var _this = this;
 
+  // In case of connection loss, running `processStalledJobs` will repick jobs here.
   return this.processStalledJobs().then(function(){
```

Comment:
Good catch:

test cases: https://github.com/OptimalBits/bull/pull/311/commits/1d8e2d2dc9f7e98996d9338ddc93498762e81c9e

fix: https://github.com/OptimalBits/bull/pull/311/commits/6e2fdf56689161a80725840f81eaf559beba4c48

That's the test case you had in mind right?